src/chunked_scatter/safe_scatter.py

```diff
@@ -0,0 +1,244 @@
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import argparse
+import math
+from typing import Generator, Iterable, List
+
+from .chunked_scatter import common_parser, region_lists_to_scatter_files
+from .parsers import BedRegion, file_to_regions
+
+
+def merge_regions(regions: Iterable[BedRegion]
+                  ) -> Generator[BedRegion, None, None]:
+    """
+    Merge regions that overlap or are exactly adjacent
+    :param regions: An iterable of possibly overlapping regions
+    :return: a generator of merged regions
+    """
+    merged_region = None
+    for region in regions:
+        if merged_region is None:
+            merged_region = region
+        else:
+            if (merged_region.contig == region.contig and
+                    merged_region.end >= region.start and
+                    region.end >= merged_region.start):
+                start = min(merged_region.start, region.start)
+                end = max(merged_region.end, region.end)
+                merged_region = BedRegion(merged_region.contig, start, end)
+            else:
+                yield merged_region
+                merged_region = region
+    if merged_region:
+        yield merged_region
+
+
+def sum_regions(regions: List[BedRegion]):
+    """ Calculate the total length of all regions """
+    return sum(len(region) for region in regions)
+
+
+def determine_bin_size(regions: List[BedRegion],
+                       scatter_count: int):
+    """
+    Determine the target scatter size, based on the total size of the regions
+    and the number of target regions (scatter_count).
+    """
+    total_size = sum_regions(regions)
+    return int(total_size/scatter_count)
+
+
+def mix_small_regions(regions, target_bin_size):
+    """ Mix small regions in between large regions
+
+        This is intended for when there are more small region than regular
+        regions. If this is not the case, we will quickly 'use up' all small
+        regions and end up with a whole bunch of uninterrupted regular regions.
+    """
+    # Small regions are regions that are smaller than the target_bin_size
+    regular_regions = [reg for reg in regions if len(reg) >= target_bin_size]
+    small_regions = [reg for reg in regions if len(reg) < target_bin_size]
+
+    # We don't want to get divide by zero errors
```

Comment:
```
try:
    small_regions_ratio = math.ceil(len(small_regions), len(regular_regions))
except ZeroDivisionError:
    small_regions_ratio = 1
```
Accomplishes the same thing and also prevents ZeroDivisionError in a more transparant way.

Reply:
Changed